phys2bids/phys2bids.py

```diff
@@ -259,13 +259,13 @@ def phys2bids(filename, info=False, indir='.', outdir='.', heur_file=None,
     if info:
         return
 
-    # Run analysis on trigger channel to get first timepoint and the time offset.
-    # #!# Get option of no trigger! (which is wrong practice or Respiract)
-    phys_in.check_trigger_amount(chtrig, thr, num_timepoints_expected, tr)
-
     # Create trigger plot. If possible, to have multiple outputs in the same
     # place, adds sub and ses label.
     if tr != 0 or num_timepoints_expected != 0:
+        # Run analysis on trigger channel to get first timepoint and the time offset.
+        # #!# Get option of no trigger! (which is wrong practice or Respiract)
+        phys_in.check_trigger_amount(chtrig, thr, num_timepoints_expected, tr)
+        thr = phys_in.thr
```

Comment:
I reply in this review because unluckily I can't reply in the general conversation.
I see what you mean, but the only other place `thr` is called now is [this line](https://github.com/physiopy/phys2bids/blob/386f3c2c51bf8774cb3dc91ad077290c90fa058d/phys2bids/phys2bids.py#L276) that is in the same `if` statement.
If I understand the program as it is now, there is no valide possible case in which `viz.plot_trigger` is not exectued without  `phys_in.check_trigger_amount(chtrig, thr, num_timepoints_expected, tr)` being executed first. `check_trigger_amount` creates the attribute that is specified here, `phys_in.thr`, no matter whether the user provides a threshold manually or not (the attribute gets created at the end of the function). So there is no valid scenario in which, when you call `viz.plot_trigger`, `phys_in.thr` doesn't exist.
It's not a major deal, so if you really want to keep this line, we can. I just don't see the reason to have it, that's all.

Reply:
Point taken. Line eliminated